themes/_core/js/product.js

```diff
@@ -172,7 +172,7 @@ function updateProduct(event, eventType, updateUrl) {
         ajax: 1,
         action: 'refresh',
         quantity_wanted:
-          eventType === 'updatedProductCombination' ? $quantityWantedInput.attr('min') : $quantityWantedInput.val(),
+          eventType === 'updatedProductCombination' ? 1 : $quantityWantedInput.val(),
```

Comment:
```suggestion
          eventType === 'updatedProductCombination' && $quantityWantedInput.attr('min') === $quantityWantedInput.val() ? 1 : $quantityWantedInput.val(),
```
I think it is better to keep the amount that the user has selected, except in the case that it is equal to the minimum amount, in which case it is understood that the user wanted the minimum amount

Reply:
hum... I'm not agree. For exemple. On a combination with minimum quantity=3 and input = 4. I change to a combinaison with minimum quantity=2 and stock=2. With your change, I have the message "not enough product in stock instead to have the new combinaison with input=2. On combination change, nothing must to be link with the input value I think because it's a kind of new product with it's own minimum quantity and stock.

Reply:
But tbh, the input should be updated to the new combination minimum order, maybe I'm wrong, but it looks like every time you'll change the combination, it will be updated to 1 even if the combination has a minimum order of 3

Reply:
No, we are speaking about the ajax request. The response always have the good minimum quantity of the new combination.

Reply:
Ah yeah, my bad just understood it

I think it's better to reset the amount in the input instead of keeping the actual amount chosen by the user, because it would lead to misunderstanding on some cases, but we can ask our @PrestaShop/prestashop-product-team :)

Reply:
@guil182  The message "not enough product in stock" is due to a bug on the PHP side.

Reply:
I think there is a PR related to this, from @atomiix

Reply:
@davidglezz I'm not sure. For example. I am on a combination with a minimum quantity of 12. I select a combination with a minimum quantity of 2 and a stock of 8.
I will have this request: ajax = 1 & action = refresh & quantity_wanted = 12

the quantity wanted sent is greater than the stock therefore the error message is "normal"

Reply:
The problem is that there are 2 errors:
1º We agree that request the minimum quantity of the previous combination as the quantity required for the new one is not correct (These are corrected here, your patch is correct in my opinion)
2º The message may be fine, but the response html is not consistent: if there is not enough quantity, it shows a message, sets the quantity input to 1 but the buy button is disabled, but it should be enabled. This error must be corrected in some PHP file

Reply:
When changing combination, the input quantity should be = to the minimal quantity defined for the combination selected.
The message "There are not enough products in stock" should be displayed when the quantity input is > to the stock, unless if ordering of out of stock product is allowed

Reply:
Hi @davidglezz,

I'm not agree with your suggestion because it's not behaviour we want ( @marionf) ->When changing combination, the input quantity should be = to the minimal quantity defined for the combination selected.

regards